openedx/core/djangoapps/certificates/api.py

```diff
@@ -42,6 +42,7 @@ def can_show_certificate_available_date_field(course):
 
 def display_date_for_certificate(course, certificate):
     if (
+        course and
```

Comment:
Doesn't this defensive line mask issues upstream?  Why should `course` ever by `NULL` when passed to this method?  If certain call paths expect a course to be `NULL`, then those can be caught within that path - without having a catch-all at this layer.

Reply:
Sure.

Reply:
Sure

Reply:
Also, it doesn't look like any of the current call paths would ever need to pass a `NULL` course through here.